enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+type PlatformOperatorStatus struct {
+    // Conditions represent the latest available observations of a platform operator's current state.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+#### Upstream OLM 1.x API(s)
+
+> Note: The following API(s) will live in the openshift/operator-framework-rukpak repository.
+
+- [Introduces the upstream Bundle and BundleDeployment OLM 1.x v1alpha1 API(s)](https://github.com/operator-framework/rukpak/tree/main/api/v1alpha1).
+- [Introduces a validating webhook for the Bundle API](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_webhook.go).
+- [Introduces a validation webhook for CRDs](https://github.com/operator-framework/rukpak/tree/main/cmd/crdvalidator).
+
+### Risks and Mitigations
+
+#### Risks
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade".
+This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
+
+We can mitigate the upstream rukpak project's direct dependency on the cert-manager operator by removing that dependency in favor of the downstream service-ca-operator component.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
```

Comment:
In phase 1, perhaps we should formalize a downstream-only `rukpak-operator` (and any others that are necessary) that can deploy rukpak (and others) based on the topology and configuration of the cluster.

This would enable things like "change the replica count to 1 if cluster is SNO" and "set these flags on the rukpak deployment based on these downstream featuregates".

Reply:
I think that's reasonable, and something we should track but it feels premature to dedicate anything significant in this proposal given it's leaning towards future phases. Maybe add this as an "answered question" and hint towards this direction?

Reply:
Added in https://github.com/openshift/enhancements/pull/1170/commits/31269f34570363358512ef7e14a4da757e779476.